### PR TITLE
Bluetooth: Fix bt_id_reset and bt_id_del without SMP

### LINKED ISSUE #73313: bt_id_reset always fails with CONFIG_BT_SMP disabled
With BT_SMP disabled: when creating a new identity with bt_id_create(NULL,NULL) and trying to rotate the private random address using bt_id_reset(<id>,NULL, NULL) the current implementation errors out because bt_unpair will always return -EINVAL

BT_SMP uses a large chunk of flash memory and some implementations may want the ability to manually rotate a static random address for privacy without having the full SMP stack.

If BT_SMP is disabled, could the easy fix just to always return success from bt_unpair since there can't be any possible pairings?

### PR DESCRIPTION
**Pull request by papatel05**
Allows BT ID resets without having the BT_SMP stack enabled. Fixes #73313

### CODE REVIEW

**Review by jori-nordic — APPROVED** [review]

**Comment by jori-nordic** [comment]
For linking the issue to the PR, you can edit the PR description and add the text below. 
That way, when the PR is merged, the issue will be automatically closed as resolved.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/73313

**Review by Thalley — APPROVED** [review]

**Comment by jhedberg** [comment]
@papatel05 thanks for submitting the pull request! The patch itself looks good, but you will need to fix some things with the commit message to make the CI Compliance Check pass. It seems like you might have used the GitHub web interface to create the PR? (based in the commit author info it looks like it). I don't think you'll be able to create a commit that satisfies CI requirements with that, rather you will need to use git from the command line. You'll e.g. need to set the git author and email properly to "First Last <email>", have a proper commit subject with a prefix ("Bluetooth: Fix bt_id_reset and bt_id_del without SMP") keep the lines at max 72 chars, and have a signed-off-by line (generated automatically when you pass -s to the git commit command). There's more information in the Zephyr documentation https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines and the links provided to you in https://github.com/zephyrproject-rtos/zephyr/pull/73374#issuecomment-2134111018.

**Comment by papatel05** [comment]
Hello @jhedberg should I start over with a new pull request?  Or could you help me modify this one to be in compliance?  I've modified against the failed checks.  I apologize for generating this one from the web interface haphazardly.

**Comment by jhedberg** [comment]
> Hello @jhedberg should I start over with a new pull request?  Or could you help me modify this one to be in compliance?  I've modified against the failed checks.  I apologize for generating this one from the web interface haphazardly.

You should be able to continue with this existing PR by doing push --force to the respective branch (patch-1). Doing simply "git show" from the command line before pushing should let you inspect that the commit message looks correct and that the author and signed-off-by line are the same and in the correct format.

**Comment by papatel05** [comment]
Hello @jhedberg, thank you for the education here.  I think I've modified the pull request in compliance now but it still says merging is blocked.  Let me know if action is required on my end.  Next pull request will be easier :)

**Review by jhedberg — APPROVED** [review]
> Hello @jhedberg, thank you for the education here. I think I've modified the pull request in compliance now but it still says merging is blocked. Let me know if action is required on my end. Next pull request will be easier :)

Thanks! I think the merging is only blocked because this still needs some approvals (I'm giving mine now) and not all CI checks have run yet. The actual commit looks correct to me now, and it's indeed always the first time when there's some overhead with getting the process right.

**Comment by Thalley** [comment]
> merging is blocked

This message appears to anyone who doesn't have merge rights :) Nothing to worry about

**Pull request merged**
